x/evm/keeper/keeper.go

```diff
@@ -305,3 +307,31 @@ func (k Keeper) AddTransientGasUsed(ctx sdk.Context, gasUsed uint64) (uint64, er
 	k.SetTransientGasUsed(ctx, result)
 	return result, nil
 }
+
+// SetHeaderHash stores the hash of the current block header in the store.
+func (k Keeper) SetHeaderHash(ctx sdk.Context) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixHeaderHash)
+	key := make([]byte, 8)
```

Comment:
```suggestion
	key := make([]byte, len(types.KeyPrefixHeaderHash)+8)
```

more efficient to build the key with prefix directly without use the prefix store.

Reply:
the idiomatic way is to create a `HeaderHashKey` function in the `x/evm/types/keys.go`, and reuse it.